### ISSUE TITLE
Stops working after Next.js Fast Refresh

### ISSUE THREAD

**Issue by nfantone** [xnimorz/use-debounce#54]
[`use-debounce`](https://github.com/xnimorz/use-debounce) stops working altogether after any changes to code are made in a Next.js project.

You can see a reproduction example here: https://codesandbox.io/s/hardcore-sara-ueu2k

Looks fine on first load, but doesn't call the input callback anymore after refresh.

### After initial load

![](http://g.recordit.co/iweO2LP6fK.gif)

### After [Fast Refresh] done 

![](http://g.recordit.co/k844RdcV1h.gif)

Some people are commenting similar situations [over at the Next.js repo](https://github.com/vercel/next.js/issues), experiencing breaking libs after they decided to rolled out their new "Fast Refresh" implementation. Oftentimes, these ended up being undiscovered bugs in the original libraries rather than a problem with Next.js itself. 

> Related Next.js issue: https://github.com/vercel/next.js/issues/13268

**Comment by xnimorz** [comment]
Hi @nfantone , @Timer 
The main tricky thing, why lots of third party libraries get crashed during fast-refresh updates, is fast-refresh re-calls (with cleanups) all `useEffects` inside custom hooks, components, etc. 

I mean, the following code wouldn't work properly:
```
function MyComponent() {
  const isComponentUnmounted = useRef(false);
  useEffect(() => () => {isComponentUnmounted.current = true}, []); 
}
```

In the example above, we change ref to the truth inside useEffect cleanup function (which is handy to prevent unnecessary timers, callbacks, etc). 
Usually, this code works correctly. But fast-refresh re-runs all useEffects, which leads to all cleanups call, and then useEffect would be called again. 

I've fixed it, by publishing a new version: `use-debounce@3.4.3`.

Here is an example of such fixes, I hope it could be helpful: https://github.com/xnimorz/use-debounce/commit/2ac17321347faf088b006c6abc2f4d332606cb91

Also, I've fixed the example by updating the version of use-debounce: https://codesandbox.io/s/hungry-architecture-4z46m?file=/pages/index.js (so that it works now)
